pilot/pkg/config/coredatamodel/syntheticserviceentrycontroller.go

```diff
@@ -101,10 +101,6 @@ func (c *SyntheticServiceEntryController) Apply(change *sink.Change) error {
 	if change.Collection != schemas.SyntheticServiceEntry.Collection {
 		return fmt.Errorf("apply: type not supported %s", change.Collection)
 	}
-	// Apply is getting called with no changes!!
```

Comment:
@Nino-K please take a look. Basically I am making Discovery service not start until everything is synced. For MCP, it is considered synced when it recieves Apply(empty change), but SSE it is not, so that was causing a deadlock. I don't think there is any reason why SSE should behave different, but wanted to verify

Reply:
I think in MCP controller we consider a config is synced when it is ran through schema validation and finally stored in the local configStore, but in sse I threw that in there as a quick validation. Ideally we should follow the same pattern  in SSE. But I think removing this is totally fine for now.